File: data/multilang_comparison/program_cpp_array/src/func26.cpp
#include "program_cpp_array.hpp" 
Array* func26(Array_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      Array* array78;
      if (pCounter > 0) {
         array78 = vars->data[--pCounter];
         array78->refC++;
         DEBUG_COPY(array78->id);
      } else {
         array78 = new Array();
         array78->size = 924;
         array78->refC = 1;
         array78->id = 78;
         array78->data = new unsigned int[array78->size]();
         DEBUG_NEW(array78->id);
      }
      unsigned int loop25 = 0;
      unsigned int loopLimit25 = (50)/3 + 1;
      for(; loop25 < loopLimit25; loop25++) {
         for (int i = 0; i < array78->size; i++) {
             array78->data[i]--;
         }
         for (int i = 0; i < array78->size; i++) {
             if (array78->data[i] == 24) {
                 return array78;
             }
         }
      }
      Array_param params0;
      params0.size = 1;
      Array* data_params0[1];
      params0.data = data_params0;
      params0.data[0] = array78;
      Array* array79 = func36(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array79->id);
      if(params0.size > 0) {
      	 params0.data.clear();
      }
      array79->refC--;
      if(array79->refC == 0) {
         delete[] array79->data;
         array79->data = nullptr;
         DEBUG_FREE(array79->id);
         delete array79;
      }
      array78->refC--;
      if(array78->refC == 0) {
         delete[] array78->data;
         array78->data = nullptr;
         DEBUG_FREE(array78->id);
         delete array78;
      }
   }
   else {
      Array_param params0;
      params0.size = 0;
      Array* data_params0[0];
      params0.data = data_params0;
      Array* array80 = func32(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array80->id);
      if(params0.size > 0) {
      	 params0.data.clear();
      }
      array80->refC--;
      if(array80->refC == 0) {
         delete[] array80->data;
         array80->data = nullptr;
         DEBUG_FREE(array80->id);
         delete array80;
      }
   }
   Array* array81;
   if (pCounter > 0) {
      array81 = vars->data[--pCounter];
      array81->refC++;
      DEBUG_COPY(array81->id);
   } else {
      array81 = new Array();
      array81->size = 379;
      array81->refC = 1;
      array81->id = 81;
      array81->data = new unsigned int[array81->size]();
      DEBUG_NEW(array81->id);
   }
   return array81;
}

File: data/multilang_comparison/program_go/src/func26.go
package main
import "math/rand"
var _ = rand.Intn
func Func26(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   if (PATH0 & 1) != 0 {
      var array78 *Array
      if pCounter > 0 {
          array78 = vars.Data[pCounter-1]
          pCounter--
          array78.RefC++
      } else {
          array78 = &Array{
              Size: 924,
              RefC: 1,
              Id: 78,
              Data: make([]uint32, 924),
          }
      }
      var loop25 int = 0;
      var loopLimit25 int = (50)/3 + 1;
      for ; loop25 < loopLimit25; loop25++ {
         for i := 0; i < array78.Size; i++ {
             array78.Data[i]--
         }
         for i := 0; i < array78.Size; i++ {
             if array78.Data[i] == 24 {
                 return array78
             }
         }
      }
      var params0 ArrayParam
      params0.Size = 1
      params0.Data = make([]*Array, params0.Size)
      params0.Data[0] = array78
      var array79 *Array = Func36(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array79.RefC--
      if array79.RefC == 0 {
          array79.Data = nil
          array79 = nil
      }
      array78.RefC--
      if array78.RefC == 0 {
          array78.Data = nil
          array78 = nil
      }
   
   } else {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array80 *Array = Func32(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array80.RefC--
      if array80.RefC == 0 {
          array80.Data = nil
          array80 = nil
      }
   }
   var array81 *Array
   if pCounter > 0 {
       array81 = vars.Data[pCounter-1]
       pCounter--
       array81.RefC++
   } else {
       array81 = &Array{
           Size: 379,
           RefC: 1,
           Id: 81,
           Data: make([]uint32, 379),
       }
   }
   return array81;
}

